dbt/adapters/base/relation.py

```diff
@@ -232,17 +233,18 @@ def quoted(self, identifier):
         )
 
     @staticmethod
-    def add_ephemeral_prefix(name: str):
-        return f"__dbt__cte__{name}"
+    def add_ephemeral_prefix(alias: str):
+        return re.sub(r"\W+", "", f"__dbt__cte__{alias}")
 
     @classmethod
     def create_ephemeral_from(
         cls: Type[Self],
         relation_config: RelationConfig,
         limit: Optional[int] = None,
     ) -> Self:
-        # Note that ephemeral models are based on the name.
-        identifier = cls.add_ephemeral_prefix(relation_config.name)
+        # Note that ephemeral models are based on the alias to give the user
+        # more control over the way that the CTE name is constructed
+        identifier = cls.add_ephemeral_prefix(relation_config.alias)
```

Comment:
There is already an [`identifier` property within `RelationConfig`](https://github.com/dbt-labs/dbt-adapters/blob/782a32b530e25afe34af6896392d9fb081dce9d1/dbt/adapters/contracts/relation.py#L55) which I believe comes from the [`alias` model config](https://docs.getdbt.com/reference/resource-configs/alias).

So I'm wondering if this small change would do the trick instead?

```suggestion
        identifier = cls.add_ephemeral_prefix(relation_config.identifier)
```

<br />

If that works, then we might be able to bypass a lot of the other proposed changes. i.e., just leave `add_ephemeral_prefix` as-is and skip the proposed regex substitution that strips certain characters.

Reply:
Nice call! I switched to `identifier` in e5c1b46807ca939548f2d503a5e6effdcc91ba86 and confirms that it works the same for both ephemeral models and unit tests in our project.